tests/test_bookmarks_updated.py

```diff
@@ -133,27 +140,33 @@ def bookmarks_test(self, conn_id, testable_streams):
-                    replication_key_value = record.get(replication_key)
-                    # verify 1st sync bookmark value is the max replication key value for a given stream
-                    self.assertLessEqual(replication_key_value, first_bookmark_value_utc, msg="First sync bookmark was set incorrectly, a record with a greater replication key value was synced")
-
-                for record in second_sync_messages:
-                    replication_key_value = record.get(replication_key)
-                    # verify the 2nd sync replication key value is greater or equal to the 1st sync bookmarks
-                    self.assertGreaterEqual(replication_key_value, simulated_bookmark_value, msg="Second sync records do not respect the previous                                                  bookmark")
-                    # verify the 2nd sync bookmark value is the max replication key value for a given stream
-                    self.assertLessEqual(replication_key_value, second_bookmark_value_utc, msg="Second sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+                # The `transactions` stream is a child of the `orders` stream. Hence the bookmark for transactions is solely dependent on the value of bookmark in 'transaction_orders' which stores the parent record's bookmark.
+                # Hence it doesn't have its own replication key.
+                if stream not in ('transactions'):
+                    replication_key = next(iter(expected_replication_keys[stream]))
+                    for record in first_sync_messages:
+                        replication_key_value = record.get(replication_key)
+                        # verify 1st sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value, first_bookmark_value_utc, msg="First sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+
+                    for record in second_sync_messages:
+                        replication_key_value = record.get(replication_key)
+                        # verify the 2nd sync replication key value is greater or equal to the 1st sync bookmarks
+                        self.assertGreaterEqual(replication_key_value, simulated_bookmark_value, msg="Second sync records do not respect the previous                                                  bookmark")
+                        # verify the 2nd sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value, second_bookmark_value_utc, msg="Second sync bookmark was set incorrectly, a record with a greater replication key value was synced")
 
                 # verify that we get less data in the 2nd sync
                 # collects has all the records with the same value of replication key, so we are removing from this assertion
```

Comment:
I think the manipulated state should be altered so this does not happen.

Reply:
Here, for the transactions stream, we had 2 bookmarks earlier i.e. transactions and transaction_orders(which stores the bookmark of the parent i.e. orders bookmark). However, as this card suggested removing the filtering of the transactions based on the transactions bookmark, we have now removed the transaction_orders completely. Hence this assertion of checking the replication key value against the bookmark value would now actually check the transaction_orders bookmark value against the replication value of the transactions. Thus, we have skipped this assertion for the stream.

Reply:
Ah I was not clear, I meant for the `collections` stream not the `transactions`. Can more data be generated, and the state injected for that stream be moved so that not less records are coming through on the second sync?

Reply:
We have generated more data for the collections stream and updated the simulated state. Hence we dont have to skip it anymore!

Reply:
Nice!